### DISCUSSION TITLE
Is there any way to include a description of a target?

### DISCUSSION THREAD

**Discussion by nsheff** [ropensci/targets#467] [Help]
When I do a `tar_meta`, one of the things I'm looking for is a list of targets I have available to load. But right now, I couldn't find a way to *annotate* or *describe* the targets like this. What I'm looking for is a way to, when I define a target, include a short textual description, and then have this show up in the `tar_meta` list. This way, I get a little bit of a context for each target so I can remember what they contain without having to load them in to inspect.

My questions are:

1. Is this possible out of the box and I just missed it?
2. If not, can you point me in a direction for how to create such functionality?

**Comment by wlandau** [comment]
I recommend R Markdown for this: https://books.ropensci.org/targets/files.html#literate-programming. I often write R Markdown reports that call `tar_read()` and explain what the targets mean. R Markdown and `targets` might get a bit more integrated in the future: https://github.com/ropensci/tarchetypes/discussions/48

**Comment by NFA** [comment]
Sorry to bump this almost two years later, but I also came looking for this functionality. So consider this as a +1 for a small user nice-to-have improvement wish. 

My pipeline for a current research article is pretty straight forward. `load_data`, `wrangle_data`, `analyze_data`, `run_model`,  `create_table`/`create_plot`, and finally render a document with quarto.  I was hoping to add some user facing documentation (high-level) during console output of `tar_make()`, as many of my colleagues without in-depth programming experience would not understand what a target is, or what building a target means. I have my `_targets.R` pretty well documented, but an end-user does not need to open this file at all.  Being able to annotate or shortly describe each target, and have that annotation being output during the build stage would be most convenient. I also have a model target (stan) that takes a while to run, so I could inform the user that the model being run will take a few minutes to complete. At its core, reproducible research is all about documentation, and it would be great to have an avenue to document what happens when running `tar_make()`.

I have currently worked around it using the solution below. But having a annotation/description parameter to tar_* functions would be much cleaner.
```
describe <- function(description) {
   info <- cli::col_cyan(cli::symbol$info)
   desc <- cli::col_white(description)
   message(paste("  ", info, desc))
}

do_test <- function() {
   describe("testing")
   do_test_()
}
do_test_ <- function() {
   [...]
}

list(
   tar_target(test, do_test())
)
```
Which produces something like:
```
here() starts at C:/Users/frenys/Papers_local/Article6
✔ skip target ts_file
✔ skip target cst_file
✔ skip target dose_file
✔ skip target doe_file
• start target ts_data
   ℹ Data wrangling: Loading raw weights. Calculating TS, and adding proper column names.
• built target ts_data [0.6 seconds]
✔ skip target cst_data
• start target dose_data
   ℹ Data wrangling: Loading dose data, adding proper column names, and marking the optimal dose.
• built target dose_data [0.2 seconds]
• start target design_of_experiment
   ℹ Data wrangling: Loading DoE matrix and adding proper column names.
• built target design_of_experiment [0.06 seconds]
• start target ts_data_summary
   ℹ Data wrangling: Imputing a missing volume measurement with a linear model to calculate TS/L. Calculating the mean and std.dev. of all TS measurements.
• built target ts_data_summary [0.3 seconds]
• start target dosage_data
   ℹ Data wrangling: Calculating the dosage based on dose and TS content.
• built target dosage_data [0.2 seconds]
• end pipeline [1.66 seconds]
```

**Reply by wlandau** [reply]
That's an interesting pattern, and the console output will continue to print successfully as long as you are running all targets locally (`tar_make()` instead of `tar_make_future()` or `tar_make_clustermq()`). Unfortunately, I do not think descriptions this long would be a great fit for R console messages. The R console messages are informal and transient, purely a matter of convenience, and I would not recommend them for serious documentation purposes. Literate programming tools like R Markdown and Quarto do the job really well.
